linker/linker_test.go

```diff
@@ -2139,4 +2217,60 @@ func TestCustomJSONNameWarnings(t *testing.T) {
 			}
 		}
 	}
+	// TODO: Need to run these test cases against protoc like other test
+	//  cases in this file. As of writing, the most recent version (v22.0)
+	//  of protoc produces too many different result with protocompile. So
+	//  we are focusing on other test cases first before protoc is fixed.
+}
+
+// testByProtoc tests the proto files with protoc. The fileNames parameter indicates the order of file
+// that should be used in the command line for protoc. fileNames can be nil when the order does not matter.
+func testByProtoc(t *testing.T, files map[string]string) (passed bool, err error) {
+	tempDir := writeFileToDisk(t, files)
+	defer os.RemoveAll(tempDir)
+
+	fileNames := make([]string, 0, len(files))
+	for fileName := range files {
+		fileNames = append(fileNames, fileName)
+	}
+	return compileByProtoc(t, tempDir, fileNames)
+}
+
+func testByProtocWithFileOrder(t *testing.T, files map[string]string, fileNames []string) (passed bool, err error) {
+	tempDir := writeFileToDisk(t, files)
+	defer os.RemoveAll(tempDir)
+	return compileByProtoc(t, tempDir, fileNames)
```

Comment:
We should verify that `len(fileNames) == len(files)` in here, to verify that caller didn't make a mistake (like a typo or inadvertent omission in `inputOrder`). We could even take a step further and also inspect the actual values to make sure they match:
```golang
for _, fileName := range fileNames {
    _, ok := files[fileName]
    require.True(t, ok)
}
```

Reply:
The assertion is added to `testByProtoc`